x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -109,7 +121,10 @@ func (k msgServer) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake)
 		return nil, errorsmod.Wrapf(types.ErrInvalidHostZone, "host zone not found in unbondings: %s", hostZone.ChainId)
 	}
 	hostZoneUnbonding.NativeTokenAmount = hostZoneUnbonding.NativeTokenAmount.Add(nativeAmount)
-	hostZoneUnbonding.UserRedemptionRecords = append(hostZoneUnbonding.UserRedemptionRecords, userRedemptionRecord.Id)
+	if !userHasRedeemedThisEpoch {
+		// Only append if a UserRedemptionRecord to the HZU if it wasn't previously appended
+		hostZoneUnbonding.UserRedemptionRecords = append(hostZoneUnbonding.UserRedemptionRecords, userRedemptionRecord.Id)
```

Comment:
(obviously out of scope of this PR/upgrade) but we should remove the URRs from the EUR/HUR. Each URR is key'd by chainId/epoch number so if we need to grab the full list for an epoch, we can just do a prefix lookup